src/main/xerus/monstercat/api/APIConnection.kt

```diff
@@ -220,6 +223,35 @@ class APIConnection(vararg path: String) : HTTPQuery<APIConnection>() {
 		fun logout() {
 			CONNECTSID.clear()
 		}
+
+		private fun convertTracklist(tracks: List<Track>) = tracks.map { HashMap<String, String>().apply { this["trackId"] = it.id; this["releaseId"] = it.release.id } }
+
+		fun editPlaylist(id: String, tracks: List<Track>? = null, name: String? = null, public: Boolean? = null, deleted: Boolean? = null) {
+			val json = HashMap<String, Any>()
```

Comment:
Please stop using HashMaps and start using objects. You can have a look at all my API code, no Hashmap anywhere.

Reply:
I can't ! I need to be able to choose which fields of the object to send, and I cannot find anything online about it. Plus bonus, the ConnectPlaylist object contains glitched out tracks.

Reply:
By glitched out tracks, I mean that the List<Track> object contains the good amount of tracks, but they're all empty and contain no data inside them. This means that I cannot send my ConnectPlaylist object back using .toString(myConnectPlaylist). And I need to be able to only include the fields I want, like only "public", only "name", etc...

Reply:
What happens if you just send the nulls to the API? Does it care or just ignore them?

Reply:
Let me check it.

Reply:
null doesn't get ignored. The API has no check on whether the sent thing is bad, corrupted, or anything at all. You can send any object like a malformed track json, and it will store it as is. So if I send null, it just stores null :(

Reply:
The HashMap approach seems like the only one viable right now... It still is easier to understand than the original string concat thing tho.

Reply:
I understand. Then we'll leave it as is :)